pages/architecture.md

```diff
@@ -33,6 +33,61 @@ Here's a diagram describing this flow:
 To see if an instrumentation library already exists for your platform, see the
 list of [existing instrumentations]({{ site.github.url}}/pages/existing_instrumentations).
 
+Example flow
```

Comment:
@adriancole This sequence diagram is easy to understand. Looks good, thank you. BTW, how did you "paint" it? What tool did you use?